### PR TITLE
Create db context, update controller and service

### PR DESCRIPTION
**Pull request by mmohamed-and**

### CODE REVIEW

File: Services/BeverageRepository.cs [hunk 1]
```diff
@@ -1,66 +1,64 @@
...
+            _dbContext = dbContext;
+        }
+        public async Task<List<Beverage>> GetAll()
+        {
+            var beverages = await _dbContext.Beverages
+                .ToListAsync();
+            return beverages;
         }
-
         public IEnumerable<Beverage> GetById(string id)
```
Comment (jimjja) [inline Services/BeverageRepository.cs]: If Id is unique you will only have 1 beverage per id. You should be returning single Beverage here

File: Services/BeverageRepository.cs [hunk 2]
```diff
@@ -1,66 +1,64 @@
...
-
-            var beverages = deserializeContent.Where(x => x.BeverageType == type);
-
+            var beverages = _dbContext.Beverages.Where(x => x.BeverageType == type);
             return beverages;
         }
+        public void AddBeverage(Beverage beverage)
+        {
+            _dbContext.Beverages.Add(beverage);
+            _dbContext.SaveChanges();
```
Comment (jimjja) [inline Services/BeverageRepository.cs]: Try to use async where possible. This way you are releasing Threads and not delaying further requests.

File: Services/BeverageRepository.cs [hunk 3]
```diff
@@ -1,66 +1,64 @@
...
+        }
+        public void UpdateBeverage(Beverage beverage)
+        {
+            if (!_dbContext.Beverages.Contains(beverage))
+            {
+                return;
+            }
+            _dbContext.Beverages.Update(beverage);
+            _dbContext.SaveChanges();
```
Comment (jimjja) [inline Services/BeverageRepository.cs]: Try to use async where possible. This way you are releasing Threads and not delaying further requests.

File: Services/BeverageRepository.cs [hunk 4]
```diff
@@ -1,66 +1,64 @@
...
+        }
+        public void DeleteBeverage(string id)
+        {
+            var beverage = _dbContext.Beverages.FirstOrDefault(x => x.Id == id);
+            if (beverage == null)
+            {
+                return;
+            }
+            _dbContext.Beverages.Remove(beverage);
+            _dbContext.SaveChanges();
```
Comment (jimjja) [inline Services/BeverageRepository.cs]: Try to use async where possible. This way you are releasing Threads and not delaying further requests.

File: ShalekKavy.Api.csproj [hunk 5]
```diff
@@ -5,8 +5,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.2" />
+    <PackageReference Include="Microsoft.CrmSdk.CoreAssemblies" Version="9.0.2.32" />
```
Comment (jimjja) [inline ShalekKavy.Api.csproj]: What is this Nuget used for?

File: Context/BeverageContext.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+namespace ShalekKavy.Api.Context
+{
+    public class BeverageContext : DbContext
+    {
+        public BeverageContext(DbContextOptions<BeverageContext> options) : base(options) { }
```
Comment (jimjja) [inline Context/BeverageContext.cs:10]: Where do you define the unique primary key for each beverage? Each item in the database needs to have a unique primary key.

**Review by jimjja — CHANGES_REQUESTED** [review]

File: ShalekKavy.Api.csproj (hunk 5 above)
Comment (mmohamed-and) [reply ShalekKavy.Api.csproj]: Removed

**Pull request merged**